src/placeholder.h

```diff
@@ -0,0 +1,9 @@
+#ifndef PLACEHOLDER_H
+#define PLACEHOLDER_H
+
+#include <dolphin/os/os.h>
+
+#define NOT_IMPLEMENTED(name)                                                  \
+    OSPanic(__FILE__, __LINE__, "#name is not implemented!");
```

Comment:
@r-burns Do you think `__func__` would work instead of `name`? I don't know how standard it is or if there are any gotchas.

Reply:
Yeah I think that would work great. You just have to be careful as it's not really a macro https://stackoverflow.com/questions/31430765/why-func-function-and-pretty-function-arent-preprocessor-macros and detecting support is a little different https://stackoverflow.com/questions/15305310/predefined-macros-for-function-name-func

Reply:
So like this?
```suggestion
    OSPanic(__FILE__, __LINE__, "%s is not implemented!", __func__);
```

Reply:
Yeah exactly